여러 기준의 상품 목록 조회 구현

Hosick opened the pull request:
<h2>상품 목록 조회 기준</h2> 
<h4>-브랜드 별 상품 조회</h4>
<h4>-정렬</h4>

- 즉시 구매가 낮은 순 정렬
- 즉시 구매가 높은 순 정렬
- 상품 발매일 순 정렬
<h4>-검색</h4>

- 상품 한글 명
- 상품 영문 명
- 상품 모델 넘버

Hosick committed:
feat: 상품 목록 조회 구현

-상품 ThumbnailResponse DTO 작성
-상품 SearchCondition 작성
-SearchProductRepository 구현
-키워드, 특정 브랜드로 검색 및 조건별 정렬 구현

issue number: #77

Hosick committed:
Merge branch 'develop' into feature/77

Hosick committed:
feat: 상품 목록 조회에 판매 최저가 추가

-상품 목록 조회에 판매 최저가 추

issue number: #77

Hosick committed:
test: 디렉토리 변경으로 인한 테스트 import 수정

-디렉토리 변경으로 인한 테스트 import 수정

issue number: #77

f-lab-dev commented on src/main/java/com/flab/shoeauction/controller/ProductApiController.java:
```diff
@@ -48,6 +52,15 @@ public void createProduct(@Valid @RequestPart SaveRequest requestDto,
+        Page<ThumbnailResponse> productThumbnailsPage =
+            productService.findProducts(condition, pageable);
+
+        return ResponseEntity.ok(productThumbnailsPage);
```
ok라면 굳이 `ResponseEntity`를 리턴할 필요는 없어보입니다~

Hosick replied:
아 넵 수정하겠습니다!

f-lab-dev commented on src/main/java/com/flab/shoeauction/controller/dto/ProductDto.java:
```diff
@@ -186,5 +188,26 @@ public WishItemResponse(Long id, Long productId, String nameKor, String nameEng,
+    }
 
+    @Getter
+    @Setter
```
여기서 setter는 무엇이 필요한걸까요?

Hosick replied:
Post 메소드에서 `@RequestBody`로 Json을 받았을 때 `Jackson2HttpMessageConverter`의 `objectMapper`를 사용해 객체로 매핑하지만, Get 메소드의 경우에는 쿼리 파라미터로 데이터를 받기 때문에  `WebDataBinder`를 사용해서 객체로 변환해줍니다.

 이 `WebDataBinder`의 기본 디폴트 바인딩 방법이 `Java Bean` 방법이기 때문에 바인딩 시 setter가 필요합니다. 하지만 Dto에 setter가 포함된다면 값이 변경되는 시점을 예상하기가 어려워지므로, `ControllerAdvice`를 선언해 `WebDatabinder`의 바인딩 방식을 `DirectFieldAccess`(필드 주입)로 변경해주는 방법이 좋을 것 같습니다!

Hosick committed:
feat: 리뷰 내용 반영

-HttpStatus가 200OK인 경우 ResponseEntity 생략

-쿼리 파라미터 바인딩 방식을 Setter 사용 방식에서 field 사용 방식으로 변경

issue number: #77
